klib/radar.c

```diff
@@ -156,21 +180,29 @@ closure_function(2, 1, status, telemetry_recv,
     } else {  /* connection closed */
         closure_finish();
         if (telemetry.dump) {
-            /* We just sent a crash report: clear the log dump (so that it's not
-             * sent again at the next boot), then send a boot event. */
-            kfunc(klog_dump_clear)();
-            deallocate(telemetry.h, telemetry.dump, sizeof(*telemetry.dump));
-            telemetry.dump = 0;
-            telemetry_boot();
+            if (telemetry.dump_done) {
+                /* We just sent a crash report: clear the log dump (so that it's not
+                 * sent again at the next boot), then send a boot event. */
+                kfunc(klog_dump_clear)();
+                deallocate(telemetry.h, telemetry.dump, sizeof(*telemetry.dump));
+                telemetry.dump = 0;
+                telemetry_boot();
+            } else {
+                telemetry_retry();
+            }
         } else if (!telemetry.running) {
-            /* The boot event has been sent: start collecting usage metrics. */
-            for (int count = 0; count < RADAR_STATS_BATCH_SIZE; count++)
-                telemetry.stats_mem_used[count] = heap_allocated(telemetry.phys);
-            telemetry_stats_send();
-            telemetry.stats_count = 0;
-            kfunc(register_timer)(CLOCK_ID_MONOTONIC, RADAR_STATS_INTERVAL, false,
-                    RADAR_STATS_INTERVAL, (timer_handler)&telemetry.stats_func);
-            telemetry.running = true;
+            if (telemetry.boot_id) {
+                /* The boot event has been sent: start collecting usage metrics. */
+                for (int count = 0; count < RADAR_STATS_BATCH_SIZE; count++)
+                    telemetry.stats_mem_used[count] = heap_allocated(telemetry.phys);
+                telemetry_stats_send();
+                telemetry.stats_count = 0;
+                kfunc(register_timer)(CLOCK_ID_MONOTONIC, RADAR_STATS_INTERVAL, false,
+                        RADAR_STATS_INTERVAL, (timer_handler)&telemetry.stats_func);
+                telemetry.running = true;kfunc(timm_dealloc)(0);
```

Comment:
These lines got joined together or maybe the timm_dealloc isn't supposed to be there?

Reply:
Oops, the timm_dealloc wasn't supposed to be there, I put it there temporarily to do a test and then forgot to remove it